bftengine/tests/bcstatetransfer/RVT_test.cpp

```diff
@@ -25,85 +25,165 @@
+  std::ostringstream getSerializedRvbData() { return rvt_->getSerializedRvbData(); }
+  bool setSerializedRvbData(std::istringstream& iss) { return rvt_->setSerializedRvbData(iss); }
+  std::string getRootCurrentValueStr() { return rvt_->getRootCurrentValueStr(); }
+  RVBId getMaxRvbId() { return rvt_->getMaxRvbId(); }
+  RVBId getMinRvbId() { return rvt_->getMinRvbId(); }
+  std::vector<RVBGroupId> getRvbGroupIds(RVBId start_block_id, RVBId end_block_id, bool flag) {
+    return rvt_->getRvbGroupIds(start_block_id, end_block_id, flag);
+  }
+  std::string getDirectParentValueStr(RVBId rvb_id) { return rvt_->getDirectParentValueStr(rvb_id); }
+  std::vector<RVBId> getRvbIds(RVBGroupId id) { return rvt_->getRvbIds(id); }
+  void printToLog() { return rvt_->printToLog(LogPrintVerbosity::SUMMARY); }
 
- private:
-  RangeValidationTree& rvt_;
+  void addNode(const RVBId start_id, const RVBId end_id);
+  void removeNode(const RVBId start_id, const RVBId end_id);
+
+ protected:
+  std::unique_ptr<RangeValidationTree> rvt_;
+  const RVTConfig config_;
+  const logging::Logger logger_;
 };
 
+void BcStTestDelegator::addNode(const RVBId start_id, const RVBId end_id) {
+  for (auto i = start_id; i <= end_id; i = i + config_.fetch_range_size_) {
+    const auto str = std::to_string(i);
+    ASSERT_NFF(rvt_->addNode(i, str.data(), str.size()));
+    ASSERT_NFF(validate());
+  }
+}
+
+void BcStTestDelegator::removeNode(const RVBId start_id, const RVBId end_id) {
+  for (auto i = start_id; i <= end_id; i = i + config_.fetch_range_size_) {
+    const auto str = std::to_string(i);
+    ASSERT_NFF(rvt_->removeNode(i, str.data(), str.size()));
+    ASSERT_NFF(validate());
+  }
+}
+
 class RVTTest : public ::testing::Test {
  public:
-  void init(RangeValidationTree& rvt) {
-    delegator_ = std::make_unique<BcStTestDelegator>(rvt);
-    logging::Logger::getInstance("concord.bft.st.rvb").setLogLevel(log4cplus::INFO_LOG_LEVEL);
-  }
-  InputValues values_{randomString(HASH_SIZE), randomString(HASH_SIZE), randomString(HASH_SIZE)};
-  std::shared_ptr<BcStTestDelegator> delegator_;
+  void SetUp() override{};
+  void TearDown() override{};
+  void init(const RVTConfig& config);
+  RVTConfig getRandomConfig() const;
+
+  std::unique_ptr<BcStTestDelegator> rvt_delegator_;
+  std::unique_ptr<DataGenerator> data_generator_;
+  RVTConfig rvt_config_{3, 4, 32};
   logging::Logger logger_{logging::getLogger("concord.bft.st.rvb")};
 };
 
-/////////////////////// starting of maths properties validation test for cryptoPP::Integer class  /////////////////////
+void RVTTest::init(const RVTConfig& config) {
+  data_generator_ = std::make_unique<DataGenerator>(config.hash_size_);
+  rvt_delegator_ = std::make_unique<BcStTestDelegator>(config, logger_);
+  logging::Logger::getInstance("concord.bft.st.rvb").setLogLevel(log4cplus::INFO_LOG_LEVEL);
+}
+
+RVTConfig RVTTest::getRandomConfig() const {
+  auto RVT_K = DataGenerator::randomNum(3, 10);
```

Comment:
I would put it 3 to 2048
This test will run thousands of times on CI after merge,  its good to get a full coverage.